packages/@biomejs/js-api/src/index.ts

```diff
@@ -227,16 +234,36 @@ export class Biome {
 	 * @param {String} content The content to lint
 	 * @param {LintContentOptions} options Options needed when linting some content
 	 */
-	lintContent(
-		content: string,
-		options: LintContentOptions,
-	): PullDiagnosticsResult {
+	lintContent(content: string, options: LintContentOptions): LintResult {
 		return this.withFile(options.filePath, content, (path) => {
-			return this.workspace.pullDiagnostics({
+			let code = content;
+
+			const { diagnostics } = this.workspace.pullDiagnostics({
 				path,
 				categories: ["Syntax", "Lint"],
 				max_diagnostics: Number.MAX_SAFE_INTEGER,
 			});
+
+			const hasErrors = diagnostics.some(
```

Comment:
I'm unsure about this part - it would be great to get some guidance from someone more familiar with the biome code base here

Reply:
Fixing a file, actually, works backwords: first you fix the file, then you pull the diagnostics.

Why? Because when fixing a file, you actually end **reducing** the diagnostics by applying the fixes. This means we need to pull the diagnostics after we fixed the file, to show the users any remaining diagnostics.

That's what we do in the CLI: https://github.com/biomejs/biome/blob/490877821cddaa508929034c32d3c547ac6d49dc/crates/biome_cli/src/execute/process_file/lint.rs#L25

Reply:
That makes perfect sense, I'll go ahead and make that change.

I'm surprised we don't need to do the same in `formatContent` [above](https://github.com/biomejs/biome/blob/6835e1d966a529186c2ad5d7201b6ec07bbe762c/packages/@biomejs/js-api/src/index.ts#L188-L227)? Though a quick test I see that I get back an `[]` for diagnostics - does the formatter simply not emit `warning` / `information` diagnostics on unformatted files?

Reply:
Formatting has different expectations and requirements.

**Usually**, formatters can't format syntax, that's why we need to check for parsing errors before running the formatter.

Same goes for broken syntax **in Biome**, broken blocks are kept verbatim, only blocks that are synaptically correct.

Given the previous statement, it's safe to assume that formatters don't emit new syntax/parse diagnostics, because there weren't any before. If they do, that's a bug in the formatter.

Reply:
I guess I naively would've expected that pulling diagnostics prior to formatting the file would've come back with the details of where it wasn't formatted correctly, and that after these issues would be gone (leaving only issues that couldn't be fixed from syntax errors, etc) - though I didn't consider that we'd just skip formatting in the presence of syntax errors, presumably to avoid making the errors worse.

Reply:
(fixed in 139c3dcea83551e4caa6454f3b4a97e792ad6c37)